pkg/performanceprofile/profilecreator/profilecreator.go

```diff
@@ -289,30 +289,108 @@ func topologyHTDisabled(info *topology.Info) *topology.Info {
+	if err != nil {
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, err
+	}
+
+	// Calculate reserved cpus.
+	// note this takes into account the offlined cpus from the step above
+	reserved, topoInfo, err := ghwHandler.GetReservedCPUs(offlined, topologyInfo, reservedCPUCount, splitReservedCPUsAcrossNUMA, disableHTFlag, htEnabled)
+	if err != nil {
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, err
+	}
+
+	// Calculate isolated cpus.
+	// Note that topology info could have been modified by "GetReservedCPUS" so
+	// to properly calculate isolated CPUS we need to use the updated topology information.
+	isolated := ghwHandler.GetIsolatedCPUs(topoInfo.Nodes, reserved, offlined)
+
+	return reserved, isolated, offlined, nil
+}
+
+func (ghwHandler GHWHandler) GetIsolatedCPUs(topologyInfoNodes []*topology.Node, reserved, offlined cpuset.CPUSet) cpuset.CPUSet {
+	return calulateIsolatedCPUSet(totalCPUSetFromTopology(topologyInfoNodes), reserved, offlined)
+}
+
+// Calculates Isolated cpuSet as the difference between all the cpus in the topology and those already choosen as reserved or offlined.
+// all cpus thar are not offlined or reserved belongs to the isolated cpuSet
+func calulateIsolatedCPUSet(total, reserved, offlined cpuset.CPUSet) cpuset.CPUSet {
+	return total.Difference(reserved.Union(offlined))
+}
+
+func (ghwHandler GHWHandler) GetOfflinedCPUs(cpuInfo *cpu.Info, topologyInfo *topology.Info, offlinedCPUCount int, disableHTFlag bool, htEnabled bool) (cpuset.CPUSet, error) {
+	offlinedCPUs := newCPUAccumulator()
+	lpOfflined := 0
+
+	//currently HT is enabled on the system and the user wants to disable HT
+	if htEnabled && disableHTFlag {
+		for _, node := range topologyInfo.Nodes {
+			lpOfflined += offlinedCPUs.AddCoresWithFilter(offlinedCPUCount, node.Cores, func(lpID int) bool { return lpID != 0 })
+		}
+	} else {
+		for _, processor := range cpuInfo.Processors {
+			if processor.NumThreads <= uint32(offlinedCPUCount-lpOfflined) {
+				// We can offline a complete CPU socket
+				//so we do it because right now its the best option for power saving
+				lpOfflined += offlinedCPUs.AddCores(allCores, processor.Cores)
+			} else {
+				// Cannot offline a complete Socket
+				// so we try to offline all the logicalProcessors but the first one on each core.
+				lpOfflined += offlinedCPUs.AddCoresWithFilter(offlinedCPUCount, processor.Cores, func(lpID int) bool { return lpID != 0 })
+			}
+		}
+	}
+
+	if lpOfflined < offlinedCPUCount {
+		log.Warnf("could not offline enough logical processors (required:%d, offlined:%d)", offlinedCPUCount, lpOfflined)
 	}
+	return offlinedCPUs.Result(), nil
+}
+
+func (ghwHandler GHWHandler) GetReservedCPUs(offlinedCPUs cpuset.CPUSet, topologyInfo *topology.Info, reservedCPUCount int, splitReservedCPUsAcrossNUMA bool, disableHTFlag bool, htEnabled bool) (cpuset.CPUSet, *topology.Info, error) {
+
+	// TODO maybe it is not a good idea to modify topoly information inside this function
```

Comment:
Altering the topology data provided by the GHW is not wrong per se, but seems a smell. However it's possible that avoiding so would require significant refactoring (I can't say, but can't rule out either)

Reply:
Yeah, that's true. I think we could revisit this after this feature is done.

Reply:
typo: `topoly`